static/app/components/modals/inviteMembersModal/inviteRowControlNew.spec.tsx

```diff
@@ -0,0 +1,138 @@
+import {TeamFixture} from 'sentry-fixture/team';
+
+import {render, screen, userEvent} from 'sentry-test/reactTestingLibrary';
+
+import {
+  defaultInviteProps,
+  InviteMembersContext,
+} from 'sentry/components/modals/inviteMembersModal/inviteMembersContext';
+import InviteRowControlNew from 'sentry/components/modals/inviteMembersModal/inviteRowControlNew';
+import TeamStore from 'sentry/stores/teamStore';
+
+describe('InviteRowControlNew', function () {
+  const teamData = [
+    {
+      id: '1',
+      slug: 'moo-deng',
+      name: "Moo Deng's Team",
+    },
+    {
+      id: '2',
+      slug: 'moo-waan',
+      name: "Moo Waan's Team",
+    },
+  ];
+  const teams = teamData.map(data => TeamFixture(data));
+
+  const getComponent = props => (
+    <InviteMembersContext.Provider value={props}>
+      <InviteRowControlNew
+        roleDisabledUnallowed={false}
+        roleOptions={[
+          {
+            id: 'member',
+            name: 'Member',
+            desc: '...',
+            minimumTeamRole: 'contributor',
+            isTeamRolesAllowed: true,
+          },
+          {
+            id: 'billing',
+            name: 'Billing',
+            desc: '...',
+            minimumTeamRole: 'contributor',
+            isTeamRolesAllowed: false,
+          },
+        ]}
+      />
+    </InviteMembersContext.Provider>
+  );
+
+  beforeEach(function () {
+    TeamStore.loadInitialData(teams);
+  });
+
+  it('renders', function () {
+    render(getComponent(defaultInviteProps));
+
+    expect(screen.getByText('Email addresses')).toBeInTheDocument();
+    expect(screen.getByText('Role')).toBeInTheDocument();
+    expect(screen.getByText('Add to team')).toBeInTheDocument();
+  });
+
+  describe.each([
+    {email: 'test-space@example.com', delimiter: ' '},
+    {email: 'test-comma@example.com', delimiter: ','},
+    {email: 'test-newline@example.com', delimiter: '{enter}'},
+  ])('updates email addresses when new emails are inputted', ({email, delimiter}) => {
+    it(`invokes the mock correctly with one using delimiter "${delimiter}"`, async () => {
+      const mockSetEmails = jest.fn();
+      render(getComponent({...defaultInviteProps, setEmails: mockSetEmails}));
+      const emailInput = screen.getByLabelText('Email Addresses');
```

Comment:
you wanted `getByRole('textarea', {name: 'Email Addresses')` since that is scoping it explicitly to only selecting input boxes